### ISSUE TITLE
Spurious `WindowEvent::Ime(Ime::Disabled)` on X11

### ISSUE THREAD

**Issue by kpreid** [rust-windowing/winit#2498]
The documentation for [`WindowEvent::Ime`](https://docs.rs/winit/0.27.3/winit/event/enum.WindowEvent.html#variant.Ime) says that

> Note: You have to explicitly enable this event using [Window::set_ime_allowed](https://docs.rs/winit/0.27.3/winit/window/struct.Window.html#method.set_ime_allowed).

But I was trying out running my application (which does not call `set_ime_allowed()`) on the Steam Deck (which uses what I am informed is a KDE desktop environment and which winit reports `is_x11()` — I'm not very familiar with Linux desktops so as to identify fine distinctions), and it reported one unexpected `WindowEvent::Ime(Ime::Disabled)` event.

The same behavior can be reproduced by taking an example from the `winit` repository (I tried `mouse_wheel`) and adding event logging so that the spurious event is detectable.

To reduce surprise, I propose that one of these changes should be made — whichever seems best:

1. `winit` should not emit any `Ime` events when `set_ime_allowed()` is false or never called.
2. The documentation should specify that these `Disabled` events (and other IME events, if applicable) may occur.
3. As option 2, and additionally, _all_ platforms should emit `Disabled` events, so that applications do not experience unnecessary platform differences.

**Comment by kchibisov** [comment]
It does everything correctly since sometimes you can't disable the IME at all. The `Disabled` is delivered to inform that it's actually not used.

**Comment by kchibisov** [comment]
>As option 2, and additionally, all platforms should emit Disabled events, so that applications do not experience unnecessary platform differences.

The disabled event is transparent to you indicating that you don't have IME running. I'm not even sure how it could cause any issue, since it removes, not enables it.

**Comment by kchibisov** [comment]
>Well, in my particular case, I assumed that since my code was not asking for IME, it would never get any IME events, and so receiving any IME events would be a bug. For another more hypothetical example, perhaps a code has some clean-up logic for Ime::Disabled events, but the code implicitly assumed that this would only be called after a preceding Enabled event, and thus hits something like an unwrap() panic (or more likely, some more complex application state mismatch) because it's trying to clean up something that doesn't exist.

Yeah, I guess I should assume that you can write faulty code. I don't really like working around the code that way, your state don't need any options at all for example...

Though, just reading through the docs it's indeed a bit of a missmatch here. Since on X11 you can't disable IME at all sometimes, but we discovered that after the docs were written.
